inference-engine/src/mkldnn_plugin/nodes/mkldnn_convert_node.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <mkldnn_extension_utils.h>
+#include "mkldnn_convert_node.h"
+#include "common/cpu_convert.h"
+#include "common/tensor_desc_creator.h"
+
+#define THROW_ERROR THROW_IE_EXCEPTION << getTypeStr() << " layer with name '" << getName() <<"' ERROR: "
+
+using namespace mkldnn;
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+
+MKLDNNConvertNode::MKLDNNConvertNode(const InferenceEngine::CNNLayerPtr& layer, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache) :
+        MKLDNNNode(layer, eng, cache) {}
+
+void MKLDNNConvertNode::getSupportedDescriptors() {
+    // if tensor descriptors are set via setDescs method we need to update the inDims/outDims data
+    // from correspond tensor descriptors.
+    if (outDims.empty() && output && output->getLayout() != InferenceEngine::Layout::ANY)
+        outDims.push_back(MKLDNNDims(output->getDims()));
+    if (inDims.empty() && input && input->getLayout() != InferenceEngine::Layout::ANY)
+        inDims.push_back(MKLDNNDims(input->getDims()));
+    if (getParentEdges().size() != 1)
+        THROW_ERROR << "Incorrect number of input edges";
+    if (getChildEdges().empty())
+        THROW_ERROR << "Incorrect number of output edges";
+}
+
+void MKLDNNConvertNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    auto layer = getCnnLayer();
+    if (layer == nullptr) {
+        THROW_ERROR << "Cannot get CNN layer";
+    }
+
+    LayerConfig config;
+    DataConfig dataIn;
+    DataConfig dataConfigOut;
+
+    config.dynBatchSupport = false;
+
+    // if input and output pointers are not null, then the inp/output tensor descriptors were set using setDescs method, so
+    // they should be used as the actual descriptors.
+    if (input && input->getLayout() != InferenceEngine::Layout::ANY && output && output->getLayout() != InferenceEngine::Layout::ANY) {
+        dataIn.desc = *input;
+        config.inConfs.push_back(dataIn);
+
+        const auto& blockingDesc = config.inConfs[0].desc.getBlockingDesc(); // inp/out layouts must be the same
+        dataConfigOut.desc = TensorDesc(output->getPrecision(), input->getDims(), blockingDesc);
+        config.outConfs.push_back(dataConfigOut);
+        supportedPrimitiveDescriptors.emplace_back(config, impl_desc_type::unknown, MKLDNNMemoryDesc(config.outConfs.front().desc).getFormat());
```

Comment:
Why cannot you just do smt like: `dataConfigOut.desc` = `dataIn.desc` and then update dataConfigOut.desc precision?